Shared/Supporting Files/Views/CategoriesView.swift

```diff
@@ -55,47 +50,42 @@ struct CategoriesView: View {
 }
 
 private extension CategoriesView {
-    struct CategoryList: View {
-        /// The samples in a category.
-        let samples: [Sample]
+    struct CategoryTile<Content: View>: View {
+        /// The name of the category.
+        let name: String
         
-        var body: some View {
-            List(samples, id: \.name) { sample in
-                NavigationLink {
-                    SampleDetailView(sample: sample)
-                        .id(sample.name)
-                } label: {
-                    SampleRow(name: AttributedString(sample.name), description: AttributedString(sample.description))
-                }
-            }
-            .listStyle(.sidebar)
-        }
-    }
-    
-    struct CategoryTitleView: View {
-        /// The category name used to load the images from assets.
-        let category: String
+        /// The destination view for the category tile to present.
+        @ViewBuilder let destination: Content
```

Comment:
Hi @philium I'd like some advice here.

Another way would be passing a closure to build the view on site such as `@ViewBuilder let destination: () -> Content`. However I struggle to think of an advantage of doing that in this particular case. Do you recommend one way over another? Thank you.

Reply:
This seems to be re-inventing the navigation link wheel a bit here. I would suggest simplifying the tile to just be the tile view and then have the tile view be the label of the navigation link:

```swift
NavigationLink {
    SampleList(samples: samples(inCategory: category)
} label: {
    CategoryTile(name: category)
}
```

Having the navigation link embedded in the category tile view seems to complicate the tile view unnecessarily.